### PR TITLE
Plugins cleanup

### LINKED ISSUE #18588: RemovePluginCommand.java reports older plugin command name
**Elasticsearch version**:
master/alpha-3

**Description of the problem including expected versus actual behavior**:
RemovePluginCommand.java error message still [reports](https://github.com/elastic/elasticsearch/blob/c257e2c51f235853c4453a86e10e463813140fc9/core/src/main/java/org/elasticsearch/plugins/RemovePluginCommand.java#L74) `plugin` as the command name,  instead of `elasticsearch-plugin`

**Steps to reproduce**:
```
abonuccelli@w530 /opt/elk/PROD/node1/elasticsearch-5.0.0-alpha3/bin $ ./elasticsearch-plugin  remove x-pack
-> Removing x-pack...
A tool for managing installed elasticsearch plugins

Commands
--------
list - Lists installed elasticsearch plugins
install - Install a plugin
remove - Removes a plugin from elasticsearch

Non-option arguments:
command              

Option         Description        
------         -----------        
-h, --help     show help          
-s, --silent   show minimal output
-v, --verbose  show verbose output
ERROR: Plugin x-pack not found. Run 'plugin list' to get list of installed plugins.
```

### PR DESCRIPTION
**Pull request by jasontedor**
This pull request is a cleanup of two issues for plugins: 
- the plugin command specified in an error message for the remove
  plugin command is incorrect
- remove the ability to specify a custom path for plugins

Closes #18588

### CODE REVIEW

**Comment by nik9000** [comment]
> remove the ability to specify a custom path for plugins

So long as we still have `readlink` on the directory then I'm ok with it. My good friend @ebernhardson uses that. Plugins are deployed to `/srv/elasticsearch/plugins` (I think) and symlinked into `/usr/share/elasticsearch/plugins` and I suspect something like that is reasonably common, especially for folks that develop their own plugins.

File: docs/reference/migration/migrate_5_0/plugins.asciidoc [hunk 1]
```diff
@@ -112,3 +112,8 @@ Previously, Java system properties could be passed to the plugin
 command by passing `-D` style arguments directly to the plugin script.
 This is no longer permitted and such system properties must be passed
 via ES_JAVA_OPTS.
+
+==== Custom plugins path
+
+The ability to specify a custom plugins path via `path.plugins` has
+been removed.
```
Comment (nik9000) [inline docs/reference/migration/migrate_5_0/plugins.asciidoc]: Maybe tell users that they can still just use a symlink if they really really want to keep things in a different place?

**Comment by nik9000** [comment]
Left a halfhearted suggestion about telling folks about symlinks in the migration docs in case they feel left out without their fancy setting.

Maybe instead of deleting the test you can change it to use a symlink for the directory? As I said I think that is reasonably common. I'm happy to come back later and do that myself if you'd prefer to merge as is.

**Comment by nik9000** [comment]
LGTM modulo those two things. Whatever you decide just let me know but I don't think this needs another review.

**Comment by nik9000** [comment]
LGTM

File: docs/reference/migration/migrate_5_0/plugins.asciidoc (hunk 1 above)
Comment (s1monw) [inline docs/reference/migration/migrate_5_0/plugins.asciidoc]: @jasontedor I think we should document the symlink option still? WDYT

**Pull request merged**